janus-idp/backstage-showcase
# chore: add patches for scaffolder audit logging with scaffolder permissions

## Description

Adding the [`scaffolder audit logging changes`](https://github.com/redhat-developer/backstage/pull/1) via a patch using `patch-package`

This will also include cherry picked changes for the scaffolder permissions since the audit logging changes are built on the `v1.26.5` branch, while the scaffolder permissions are released in `v1.28.0-next.1`

Relates to https://github.com/janus-idp/backstage-showcase/pull/1265

## Which issue(s) does this PR fix

Relates to [RHIDP-2324](https://issues.redhat.com/browse/RHIDP-2324)

## PR acceptance criteria

Please make sure that the following steps are complete:


## How to test changes / Special notes to the reviewer
If running on Node 20, start with:
```bash
NODE_OPTIONS=--no-node-snapshot
```

Verifying audit logging
  - Use the scaffolder (frontend or backend) (make sure you are logged in with a backstage identity or else `actorId` will be `null`)
  - Observe audit logs be outputted for router endpoint access, task creation, cancellation, execution, etc.

The new permissions add the following new behaviours:
- Verify that new permissions are available via `<BACKEND_URL>/api/scaffolder/.well-known/backstage/permissions/metadata`
- `scaffolder.task.read`: prevents reading of tasks, task events/eventstreams
  - Test with the `GET /v2/tasks`, `GET /v2/tasks/:taskId`, `GET /v2/tasks/:taskId/events` and `GET /v2/tasks/:taskId/eventstream` endpoints
  - Test with the UI: View log stream, list tasks
- `scaffolder.task.create`: prevents creation of scaffolder tasks (can't trigger a software template)
  - Test with the `POST /v2/tasks` endpoint and try launching a software template via the UI (catalog entity about page, and Create page should both have their buttons disabled)
- `scaffolder.task.cancel`: prevents cancellation of scaffolder tasks
  - Test with the `POST /v2/tasks/:taskId/cancel` endpoint and try cancelling a running task in the UI (cancel button should be disabled)
